storage/block/types.go

```diff
@@ -169,12 +172,13 @@ type DatabaseBlock interface {
 	// from storage to serve as a memory cached block for reads.
 	IsCachedBlock() bool
 
-	// Reset resets the block start time and the segment.
-	Reset(startTime time.Time, segment ts.Segment)
+	// Reset resets the block start time, duration, and the segment.
+	Reset(startTime time.Time, blockSize time.Duration, segment ts.Segment)
```

Comment:
maybe this should just take a `xio.Block` instead?

Reply:
I think this gets a little messy in that a lot of the callers of Reset() would probably want to avoid building the full BlockReader? If you think otherwise I can make the change